packages/strapi-admin/ee/admin/containers/Roles/CreatePage/ContentTypeCollapse/CollapsePropertyMatrix/SubActionRow/index.js

```diff
@@ -0,0 +1,116 @@
+import React, { memo, useCallback, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Flex, Text } from '@buffetjs/core';
+import styled from 'styled-components';
+import CheckboxWithCondition from '../../../CheckboxWithCondition';
+import Chevron from '../../../Chevron';
+import CollapseLabel from '../../../CollapseLabel';
+import Curve from '../../../Curve';
+import HiddenAction from '../../../HiddenAction';
+import RequiredSign from '../../../RequiredSign';
+import { RowStyle, RowWrapper } from './row';
+import { LeftBorderTimeline, TopTimeline } from './timeline';
+import Wrapper from './Wrapper';
+
+const SubLevelWrapper = styled.div`
+  padding-bottom: 8px;
+`;
+
+const SubActionRow = ({ recursiveLevel, values, propertyActions }) => {
+  const [rowToOpen, setRowToOpen] = useState(null);
+  const handleClickToggleSubLevel = useCallback(name => {
```

Comment:
Not sure useCallback is necessary: this function is used in an arrow function, the child component will render anyways.

Reply:
Fixed in #9405